### DISCUSSION TITLE
epiPool: tools for estimating/confirming outbreak prevalence via pool testing approach

### DISCUSSION THREAD

**Discussion by abdoelnaser-mahmood-degoot** [epiverse-trace/epiverse-trace.github.io#60] [Project Ideas]
### Description

This package would provide methods to quickly, cheaply and accurately confirm or estimate the prevalence of an outbreak.The scale and scope of the measures taken to control the spread of an outbreak are largely determined by the prevalence level of the outbreak in the community of interest.
The basic approach to estimating the outbreak prevalence is to randomly choose n ≫ 1 individuals from a much larger population of interest, test each of them, count how many are positive, say k, and then divide this number by n to get the expected prevalence as p = k/n with variance ≈ k/n. The number of tests, n, is determined by a specified level of accuracy, δp - larger values of n are required for greater precision. For example, suppose that preliminary clinical data indicate that a circulating outbreak has a prevalence of around p0 ≈ 0.1%, and we wish to confirm this to a precision level of δp < 0.1%, so that we are 99% confident that the prevalence is less than 0.2%. This requires over 1000 tests to be carried out, which takes a lot of resources ( both manpower and money). In addition, by the time all these tests are completed, the actual value of p0 in the community may have changed. Interventions implemented on the basis of the previous value of p0, confirmed (or adjusted) by the analysis of the tests, may be less effective.

The ability to estimate prevalence with fewer tests could save time and resources. It could potentially help control outbreaks more quickly. The vast majority of individuals will be negative at low prevalence levels, as is the case in initial stages of outbreaks. However, the pooled testing approach tells us that many negative individuals can be detected with a single test - assuming that a pool test will be positive if and only if at least one individual in the pool has the infection. The overall results of a group testing experiment (the number of positive and negative groups) provide sufficient information to estimate prevalence. For example, in the above scenario, a comparable level of confidence could be achieved with fewer than 40 tests.

Overall, the proposed package would provide reliable methods for estimating the prevalence of an outbreak. This could help public health officials make informed decisions about how to control the spread of the outbreak and allocate resources effectively.

### Typical end-users

* field epidemiologists and policy makers

### Potential contributors

* Individuals from groupool (https://www.groupool-covid19.org/) *members of the Epiverse-TRACE team, as well as others who have worked on pool testing approaches and applications. 

### Key collaborators

* research groups with similar interest

### Inputs

* none

### Outputs
* data.frame or tibble describing the group testing experiment layout *Shiny app

### Imports
* random
* stats
* optima
* data.table

### Used by

* none

### Related projects
* epiR 
* prevalence 

### Usage

```bash
library(epipool)
library(aptimr)
library(random)
library(stats)
# Use case 1: calculate the optimal number of tests given p0 and δp
calculate_optimum_number_tests(p0, δp = 0.1%)
# Use case 2: generate an optimal group testing layout given p0 and n
generate_experiment_layout(n, p0)
# Use case 3: estimate the prevalence level
estimate_prevalence(#groups, #positive/negative groups)
```

### Additional comments
Hello to everyone. As you may be aware, I am new to this business. Please provide feedback on the above concept. Your comments will help me shape it better or discard it entirely !

**Comment by adamkucharski** [comment]
Thanks for sharing. A few thoughts on existing methods/tools, and potential links:

- Pool testing is definitely a use case that crops up quite often - we've previously thought about pooled testing for travel testing data, albeit with quite crude functions (https://www.medrxiv.org/content/10.1101/2022.10.12.22280928v1).

- There’s also this package that could be useful starting point (I'm collaborating on other projects with some of the authors, so can make connection if useful): https://www.sciencedirect.com/science/article/pii/S1364815221002012

- There's also @avallecam 's package on seroprevalence estimation, which might have some useful functionality for false positives/negatives: https://avallecam.github.io/serosurvey/

- [serofoi](https://github.com/epiverse-trace/serofoi) has functionality for seroprevalence data, so perhaps there is a broader data structure we'd want for test data more generally, then could pass to different packages (e.g. serofoi, this one). Would be great to find ways to address underlying gaps in the Epiverse ecosystem, as well as this specific question about pooling.

**Reply by adamkucharski** [reply]
Adding this package, which looks to cover key functions: https://github.com/CDCgov/PooledInfRate
